### DISCUSSION TITLE
log storage

### DISCUSSION THREAD

**Discussion by downdawn** [fastapi-practices/fastapi_best_architecture#49] [Ideas]
Any ideas on log storage?

There are various strategies for handling large amounts of log data.

**Comment by wu-clan** [comment]
What do you have in mind?

**Reply by downdawn** [reply]
Common logs include operation logs, login logs, and exception logs

Most commonly, add a background task in AccessMiddleware to store log information, such as directly to a mysql database.

There are other solutions, such as ELK.

**Reply by wu-clan** [reply]
ELK is too heavy

1. distinguish error log and output log to different files, need to add records manually in the code
2. create system operation log table, record operation log, including operator, time, access interface... 
3. create login log table, record user login logs, including name, time, IP...

**Reply by downdawn** [reply]
ok

**Comment by wu-clan** [comment]
- [x] separate log files (stdout, stderr) #77

- [x] add login logs #76

- [x] add opera logs #92
